rotkehlchen/tests/unit/test_tasks_manager.py

```diff
@@ -221,21 +222,37 @@ def test_check_premium_status(rotkehlchen_api_server):
     premium = Premium(premium_credentials)
     premium.status = SubscriptionStatus.ACTIVE
 
+    def mock_check_premium_status():
+        task_manager.last_premium_status_check = ts_now() - PREMIUM_STATUS_CHECK
+        task_manager._maybe_check_premium_status()
+
     with patch(
         'rotkehlchen.db.dbhandler.DBHandler.get_rotkehlchen_premium',
         MagicMock(return_value=premium_credentials),
     ):
         assert premium.is_active() is True
         assert rotki.premium is not None
 
-        timeout = 5
-        try:
-            with gevent.Timeout(timeout):
-                task_manager.schedule()
-        except gevent.Timeout as e:
-            raise AssertionError(f'Premium check query was not completed within {timeout} seconds') from e  # noqa: E501
+        with patch('rotkehlchen.premium.premium.Premium.is_active', MagicMock(return_value=False)):
+            for check_trial in range(3):
+                mock_check_premium_status()
+                assert rotki.premium is not None, (
+                    'Premium object is None and should be'
+                    f'active after the {check_trial} periodic check'
+                )
 
-        assert rotki.premium is None
+            mock_check_premium_status()
+            assert rotki.premium is None, (
+                'Premium object is not None and should be'
+                'deactivated after the 4th periodic check'
+            )
```

Comment:
just as a comment when the line is near the max-length allowed to avoid adding multiple short lines we use  `  # noqa: E501` that skips the lint check